components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -489,8 +490,11 @@ class GeckoEngineSession(
         geckoSession.contentDelegate = createContentDelegate()
         geckoSession.trackingProtectionDelegate = createTrackingProtectionDelegate()
         geckoSession.permissionDelegate = createPermissionDelegate()
+        geckoSession.promptDelegate = createPromptDelegate()
     }
 
+    private fun createPromptDelegate() = GeckoPromptDelegate(this)
```

Comment:
Do we need this method or could we create the delegate in `initGeckoSession` directly? Afaik the other create() methods only exist because we are creating all the instances inline and that separates the code.

Reply:
You are right. First I did the inline implementation but it was to much to be in the same class, I just forgot about `createPromptDelegate()`